lib/components/map/bounds-updating-overlay.js

```diff
@@ -69,7 +69,31 @@ class BoundsUpdatingOverlay extends MapLayer {
     const oldIntermediate = oldProps.query && oldProps.query.intermediatePlaces
     const newIntermediate = newProps.query && newProps.query.intermediatePlaces
     const intermediateChanged = !isEqual(oldIntermediate, newIntermediate)
-    if (
+
+    // Also refit map if itineraryView prop has changed.
+    const itineraryViewChanged = oldProps.itineraryView !== newProps.itineraryView
+
+    if (itineraryViewChanged) {
```

Comment:
This is perhaps the wrong place for this comment, but when you click a leg summary to show the split screen map view, is there any way to go back to the full itinerary view?

Reply:
Good idea to insist. In https://github.com/opentripplanner/otp-react-redux/pull/339/commits/7678132763b027dc21ff3d7784230b7fe5577913, clicking on the same leg in the narrative will toggle between the split map and the full itinerary view.